test/test_nn.py

```diff
@@ -11128,32 +11128,39 @@ def test_softmax_bfloat16(self, device):
     @onlyCUDA
     @skipCUDAIfRocm
     @skipCUDAIfCudnnVersionLessThan(7603)
-    def test_conv_cudnn_nhwc(self, device):
-        input = torch.randint(1, 10, (2, 8, 4, 4), dtype=torch.float32, device=device, requires_grad=True)
-        input = input.contiguous(memory_format=torch.channels_last)
-        input.retain_grad()
-        grad = torch.rand(2, 4, 2, 2, dtype=torch.float32, device=device)
-        grad = grad.contiguous(memory_format=torch.channels_last)
-        conv = nn.Conv2d(8, 4, 3).cuda().float()
-        conv.weight.data = conv.weight.contiguous(memory_format=torch.channels_last)
-
-        ref_input = input.detach().clone().contiguous().requires_grad_(True)
-        ref_grad = grad.detach().clone().contiguous()
-        ref_conv = nn.Conv2d(8, 4, 3).cuda().float()
-        # load_state_dict will restore the stride & memory_layout on ref_conv.weight.
-        ref_conv.load_state_dict(conv.state_dict())
+    @dtypes(torch.half, torch.float)
+    def test_conv_cudnn_nhwc(self, device, dtype):
+        def _helper(n, c, h, w, out_channels, kernel_size, groups):
+            input = torch.randn(n, c, h, w, dtype=dtype, device=device).to(memory_format=torch.channels_last)
+            input.requires_grad_()
+            conv = nn.Conv2d(c, out_channels, kernel_size, groups=groups).cuda().to(dtype).to(memory_format=torch.channels_last)
+
+            ref_input = input.detach().clone().contiguous().requires_grad_()
+            ref_conv = nn.Conv2d(c, out_channels, kernel_size, groups=groups).cuda().to(dtype)
+            # load_state_dict will restore the stride & memory_layout on ref_conv.weight.
+            ref_conv.load_state_dict(conv.state_dict())
+            ref_conv = ref_conv.to(memory_format=torch.contiguous_format)
+
+            out = conv(input)
+            ref_out = ref_conv(ref_input)
+
+            grad = torch.randn_like(out)
+            ref_grad = grad.detach().clone().contiguous()
 
-        out = conv(input)
-        out.backward(grad)
-        ref_out = ref_conv(ref_input)
-        ref_out.backward(ref_grad)
+            out.backward(grad)
+            ref_out.backward(ref_grad)
 
-        self.assertTrue(out.is_contiguous(memory_format=torch.channels_last))
-        self.assertTrue(ref_out.is_contiguous())
-        self.assertEqual(out, ref_out)
-        self.assertEqual(conv.weight.grad, ref_conv.weight.grad)
-        self.assertEqual(conv.bias.grad, ref_conv.bias.grad)
-        self.assertEqual(input.grad, ref_input.grad)
+            self.assertTrue(out.is_contiguous(memory_format=torch.channels_last))
+            self.assertTrue(ref_out.is_contiguous())
+            self.assertEqual(out, ref_out, rtol=0.001, atol=0.001)
+            self.assertEqual(conv.weight.grad, ref_conv.weight.grad, rtol=0.001, atol=0.001)
+            self.assertEqual(conv.bias.grad, ref_conv.bias.grad, rtol=0.001, atol=0.001)
+            self.assertEqual(input.grad, ref_input.grad, rtol=0.001, atol=0.001)
```

Comment:
Could we use the `dtype_precisions` from [here](https://github.com/pytorch/pytorch/blob/9cfc10d52e0d0a8576b0a5a347fa6fa8da86244a/torch/testing/_internal/common_utils.py#L887-L895) for the check?

Reply:
Thanks! Will do!